src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -52,6 +55,7 @@ def __init__(
         self._seen_tables: dict[str, str] = {}
         self._migrate_grants = migrate_grants
         self._external_locations = external_locations
+        self._spark = SparkSession.builder.getOrCreate()
```

Comment:
please isolate spark session creation only to the method that uses it. See this PR https://github.com/databrickslabs/ucx/pull/2947/files

Reply:
i have moved this to a seperate property called _spark_session
but I am now calling this in migrate_tables so that the spark session instantiate when the value passed is "CONVERT_TO_EXTERNAL". also doing it at migrate_tables allows to create the session just once. calling it in child method would invoke this for each table migration.